docs/git/cheat-sheet.md

```diff
@@ -0,0 +1,67 @@
+# Git Cheat Sheet
+
+The following is a cheat sheet for using [git] while developing Cardano SL
+which should contain all the git operations that someone working on Cardano
+will need at some point.
+
+### Cloning the Cardano SL repository
+```
+git clone https://github.com/input-output-hk/cardano-sl
+```
+After you change directory into that cloned repository, you may want to tell
+git to use the SSH protocol for the "git push" command, keeping the HTTPS
+protocol for the "git pull" and "git fetch" commands by issuing the following
+command:
+
+```
+git remote set-url --push origin git@github.com:input-output-hk/cardano-sl
+```
+
+### Listing the current and local branches
+```
+git branch
+```
+To list all the branches both local and at the origin (usually Github):
+```
+git branch --all
+```
+
+### Create and change into a new branch
+Development on Cardano SL is usually done by branching off the "develop" branch.
+Assuming you are already on the develop branch and your user name is "xyz" and
+you are working on YouTrack ticket number "YT-1234":
+```
+git branch xyz/YT-1234
+git checkout xyz/YT-1234
+```
+A shorter version of the above is:
+```
+git checkout -b xyz/YT-1234
+```
```

Comment:
Perhaps add a small note: 

> This can safely be done at any time. If a branch with a same name already exists, git will warn you about it and deny the command. One can forcefully override an existing branch using a capital `-B` as an option:
> 
> ```
> git checkout -B xyz/YT-1234
> ```
> 
> As a matter of fact, a lot of git options are applied safely when used in their lowercase form, and forcefully when used in their uppercase form.

Reply:
I did know about that one, and chose the lower case one :)

Reply:
Yes, I do prefer the lowercase too (safer). The paragraph is just about giving people a sense of the `git` philosophy towards options (`-m` / `-M`, `-b` / `-B` etc..).

Reply:
@disassembler Well, it is rather good to have a "branch owner" with our current workflow, especially since we're rebasing quite often stuff. I think we all expect to be the only one working on a branch 99% of the time, and when one has to push to someone's else branch, I believe it's good to ask before or let the person know (to avoid overwriting stuff after a forced push). 

Hence the name + ticket nomenclature. It makes it clear whom one should ask about the branch.